src/cmd/tag.go

```diff
@@ -120,28 +81,25 @@ opslevel get tag --type=Service ID|ALIAS KEY | jq
 		err := validateResourceTypeArg()
 		cobra.CheckErr(err)
 
-		resource := args[0]
+		resourceIdentifier := args[0]
 		tagKey := args[1]
 
-		var result any
-		if opslevel.IsID(resource) {
-			id := opslevel.ID(resource)
-			result, err = TaggableResourceFetchFunctions[opslevel.TaggableResource(resourceType)](id)
-		} else {
-			alias := args[0]
-			result, err = TaggableResourceFetchAliasFunctions[opslevel.TaggableResource(resourceType)](alias)
-		}
-
-		tags, err := GetTags(result)
+		client := getClientGQL()
+		result, err := client.GetTaggableResource(opslevel.TaggableResource(resourceType), resourceIdentifier)
 		cobra.CheckErr(err)
 
-		output := []opslevel.Tag{}
+		tags, err := result.GetTags(client, nil)
+		cobra.CheckErr(err)
+		foundTag := opslevel.Tag{}
 		for _, tag := range tags.Nodes {
-			if tagKey == tag.Key {
-				output = append(output, tag)
+			if tag.Key == tagKey {
+				foundTag = tag
+				break
```

Comment:
ahh ya @taimoor-at-opslevel is right we should not "break" here and instead build up a list of tags that match the key.